modes/magit/evil-collection-magit.el

```diff
@@ -39,13 +55,633 @@
 (defconst evil-collection-magit-maps '(magit-blame-mode-map
                                        magit-blame-read-only-mode-map))
 
+(defcustom evil-collection-magit-use-y-for-yank t
+  "When non nil, replace \"y\" for `magit-show-refs-popup' with
+\"yy\" for `evil-yank-line', `ys' `magit-copy-section-value',
+\"yb\" for `magit-copy-buffer-revision' and \"yr\" for
+`magit-show-refs-popup'. This keeps \"y\" for
+`magit-show-refs-popup' in the help
+popup (`magit-dispatch-popup'). Default is t."
+  :group 'magit
+  :type 'boolean)
+
+(defcustom evil-collection-magit-want-horizontal-movement nil
+  "When non nil, use \"h\" and \"l\" for horizontal movement in
+most magit buffers. The old \"h\" for help is moved to \"H\" and
+similarly for the old \"l\" for the log popup. The old \"L\" is
+then put on \"C-l\"."
+  :group 'magit
+  :type  'boolean)
+
+(defcustom evil-collection-magit-use-z-for-folds nil
+  "When non nil, use \"z\" as a prefix for common vim fold commands, such as
+  - z1 Reset visibility to level 1 for all sections
+  - z2 Reset visibility to level 2 for all sections
+  - z3 Reset visibility to level 3 for all sections
+  - z4 Reset visibility to level 4 for all sections
+  - za Toggle a section
+  - zo Show section
+  - zO Show sections recursively
+  - zc Hide section
+  - zC Hide sections recursively
+  - zr Same as z4.
+
+When this option is enabled, the stash popup is available on \"Z\"."
+  :group 'magit
+  :type  'boolean)
+
+(defcustom evil-collection-magit-state (if evil-collection-magit-use-y-for-yank 'normal 'motion)
+  "State to use for most magit buffers."
+  :group 'magit
+  :type  'symbol)
+
+;; without this set-mark-command activates visual-state which is just annoying
+;; and introduces possible bugs
+(defun evil-collection-magit-remove-visual-activate-hook ()
+  (when (derived-mode-p 'magit-mode)
+    (remove-hook 'activate-mark-hook 'evil-visual-activate-hook t)))
+(add-hook 'evil-local-mode-hook 'evil-collection-magit-remove-visual-activate-hook)
+
+(defun evil-collection-magit-maybe-deactivate-mark ()
+  "Deactivate mark if region is active. Used for ESC binding."
+  (interactive)
+  (when (region-active-p) (deactivate-mark)))
+
+(defun evil-collection-magit-define-key (state map-sym key def)
```

Comment:
I didn't look at the delta between this and evil-collection-define-key but I'm guessing evil-collection-define-key is needed for the key translation, definition, etc code.

Reply:
I think we can throw away the one from evil-magit and use the evil-collection one

Reply:
Sounds good, removed.